src/buffer/manager.rs

```diff
@@ -34,6 +34,9 @@ pub trait BufferMgr {
     fn flush_all(&mut self, txnum: i32) -> Result<()>;
     fn unpin(&mut self, buff: Arc<Mutex<Buffer>>) -> Result<()>;
     fn pin(&mut self, blk: &BlockId) -> Result<Arc<Mutex<Buffer>>>;
+    // extends for statistics by exercise 4.18
+    fn nums_total_pinned_unpinned(&self) -> (u32, u32);
+    fn buffer_cache_hit_ratio(&self) -> f32;
```

Comment:
Is this name wrong?
Correctly speaking, this means a ratio between cache hit and cache miss!